caramel/compiler/ocaml_to_erlang/fun.ml

```diff
@@ -261,7 +263,21 @@ and mk_expression exp ~var_names ~modules ~functions ~module_name =
                   (mk_expression arg ~var_names ~modules ~functions ~module_name))
           args
       in
-      Expr.apply name args
+      if wanted_arity > List.length args then
+        let rec gen_names acc n =
+          match n with
+          | 0 -> acc
+          | n ->
+              gen_names
+                (Name.var ("Caramelinternal" ^ string_of_int n) :: acc)
```

Comment:
Can we use `"Caramel@arg"` here? The `@` is a valid variable character, and it will give a nice sense of "namespacing" to these names.